openmdao/drivers/pyoptsparse_driver.py

```diff
@@ -169,43 +204,150 @@ def run(self, problem):
         self.quantities += list(econs)
 
         for name in self.get_constraints(ctype='eq'):
-            size = con_meta[name]['size']
-            lower = upper = con_meta[name]['equals']
+            meta = con_meta[name]
+            size = meta['size']
+            lower = upper = meta['equals']
 
             # Sparsify Jacobian via relevance
-            wrt = rel.relevant[name].intersection(indep_list)
+            rels = rel.relevant[name]
+            wrt = rels.intersection(indep_list)
             self.sparsity[name] = wrt
 
-            if con_meta[name]['linear'] is True:
+            if meta['linear'] is True:
                 opt_prob.addConGroup(name, size, lower=lower, upper=upper,
                                      linear=True, wrt=wrt,
                                      jac=self.lin_jacs[name])
             else:
+
+                jac = None
+
+                # Additional sparsity for index connections
+                for param in wrt:
+
+                    sub_conns = sub_param_conns.get(param)
+                    if not sub_conns:
+                        continue
+
+                    # If we have a simultaneous full connection, then we move on
+                    full_conns = full_param_conns.get(param)
+                    if full_conns.intersection(rels):
+                        continue
+
+                    rel_idx = set()
+                    for target, idx in iteritems(sub_conns):
+
+                        # If a target of the indexed desvar connection is
+                        # in the relevant path for this constraint, then
+                        # those indices are relevant.
+                        if target in rels:
+                            rel_idx.update(idx)
+
+                    nrel = len(rel_idx)
+                    if nrel > 0:
+
+                        if jac is None:
+                            jac = {}
+
+                        if param not in jac:
```

Comment:
This isn't a huge deal, but couldn't you build the COO matrix directly above around line 178?